sdk/core/core-rest-pipeline/src/nodeHttpClient.ts

```diff
@@ -243,28 +246,55 @@ class NodeHttpClient implements HttpClient {
   }
 
   private getOrCreateAgent(request: PipelineRequest, isInsecure: boolean): http.Agent {
-    if (!request.disableKeepAlive) {
-      if (isInsecure) {
-        if (!this.httpKeepAliveAgent) {
-          this.httpKeepAliveAgent = new http.Agent({
-            keepAlive: true,
-          });
-        }
+    const disableKeepAlive = request.disableKeepAlive;
 
-        return this.httpKeepAliveAgent;
-      } else {
-        if (!this.httpsKeepAliveAgent) {
-          this.httpsKeepAliveAgent = new https.Agent({
-            keepAlive: true,
-          });
-        }
+    // Handle Insecure requests first
+    if (isInsecure) {
+      if (disableKeepAlive) {
+        // keepAlive:false is the default so we don't need a custom Agent
+        return http.globalAgent;
+      }
 
-        return this.httpsKeepAliveAgent;
+      // Get the cached agent if exists
+      let agent = this.cachedHttpAgent;
+      if (!agent) {
+        // If there is no cached agent create a new one and cache it.
+        agent = new http.Agent({ keepAlive: true });
+        this.cachedHttpAgent;
       }
-    } else if (isInsecure) {
-      return http.globalAgent;
+      return agent;
     } else {
-      return https.globalAgent;
+      // We use the tlsSettings to index cached clients
+      const tlsSettings = request.tlsSettings ?? DEFAULT_TLS_SETTINGS;
```

Comment:
for clarity I'd move this below the next if() block, since we don't need the `tlsSettings` variable until after we verify that we have some settings on the request